lib/Amazon/S3/Thin/Credentials.pm

```diff
@@ -20,13 +20,28 @@ Amazon::S3::Thin::Credentials - AWS credentials data container
 
 =head1 DESCRIPTION
 
-This module contains aws credentials and provide getters to the data.
+This module contains AWS credentials and provide getters to the data.
+
+    # Load from arguments
+    my $creds = Amazon::S3::Thin::Credentials->new($access_key, $secret_key, $session_token);
+
+    # Load from environment
+    my $creds = Amazon::S3::Thin::Credentials->from_env
+
+    # Load from instance profile
+    my $creds = Amazon::S3::Thin::Credentials->from_instance(role => 'foo', version => 2);
 
 =cut
 
 use strict;
 use warnings;
 
+use Carp;
+use JSON::PP;
+use LWP::UserAgent;
+
+my $JSON = JSON::PP->new->utf8->canonical;
```

Comment:
This variable is never used, you are probably using decode_json instead and I recommend to write `use JSON::PP ();` to avoid using exported subroutines.